### PR TITLE
A couple fixes to make Docker For Mac work

### PR DESCRIPTION
**Pull request by evanphx**
This should fix https://github.com/hashicorp/nomad/issues/1080 as well as https://github.com/hashicorp/nomad/issues/1091

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -275,6 +275,12 @@ func (c *Client) init() error {
 		if err != nil {
 			return fmt.Errorf("failed creating temporary directory for the StateDir: %v", err)
 		}
+
+		p, err = filepath.EvalSymlinks(p)
+		if err != nil {
+			return err
```
Comment (sethvargo) [inline client/client.go]: I believe we should prefix this error with something so we can trace it better (see other errors for examples)

Comment (evanphx) [inline client/client.go]: Fixed!

**Comment by evanphx** [comment]
This PR currently triggers a panic: https://gist.github.com/evanphx/e3883d68062ff4c67ee0b6e5633df077

**Comment by evanphx** [comment]
Experiencing another crash: https://gist.github.com/evanphx/48a594053af828e9179169bc724a36c3

**Comment by jippi** [comment]
any love for this PR before 0.5.0 ? :)

**Comment by sethvargo** [comment]
Hi @jippi 

Thank you for your comment. We are currently investigating why this PR is causing an upstream panic and crash. Until we can resolve that, we won't be able to get this into a release. Thanks! 😄

**Comment by jippi** [comment]
yay @dadgar !

**Pull request merged**